test/src/test/java/jenkins/model/GlobalBuildDiscarderTest.java

```diff
@@ -72,9 +75,10 @@ public void testJobBuildDiscarder() throws Exception {
             Assert.assertArrayEquals("newest 2 builds", p.getBuilds().stream().mapToInt(Run::getNumber).toArray(), new int[]{8, 7});
             j.buildAndAssertSuccess(p);
             j.buildAndAssertSuccess(p);
+            GlobalBuildDiscarderListener.await();
 
             // run global discarders once a build finishes
-            Assert.assertArrayEquals("2 builds because of BackgroundBuildDiscarderListener", p.getBuilds().stream().mapToInt(Run::getNumber).toArray(), new int[]{10, 9});
+            Assert.assertArrayEquals("2 builds because of GlobalBuildDiscarderListener", p.getBuilds().stream().mapToInt(Run::getNumber).toArray(), new int[]{10, 9});
```

Comment:
`BackgroundBuildDiscarderListener` does not exist, so I suppose this was meant. Seems to make more sense in this context than `BackgroundGlobalBuildDiscarder`.